modules/gcp/win-gfx/win-gfx-startup.ps1.tmpl

```diff
@@ -224,8 +253,21 @@ function Join-Domain {
         # - when password is incorrect (retry because user might not be added yet)
         # - when computer is already in domain
         Catch [System.InvalidOperationException] {
-            $_.Exception.Message
-            if (($Elapsed -ge $Timeout) -or ($_.Exception.GetType().FullName -match "AddComputerToSameDomain,Microsoft.PowerShell.Commands.AddComputerCommand")) {
+            $PSItem
+
+            # Sometimes domain join is successful but renaming the computer fails

```

Comment:
we don't rename the computer in GCP, unlike in AWS.  Please remove rename-related handling. See line 248 vs in AWS where we have the "-NewName "$host_name"" option.